src/evaluators/interpolation/PHAL_NodesToCellInterpolation_Def.hpp

```diff
@@ -13,10 +13,25 @@ namespace PHAL {
 template<typename EvalT, typename Traits, typename ScalarT>
 NodesToCellInterpolationBase<EvalT, Traits, ScalarT>::
 NodesToCellInterpolationBase (const Teuchos::ParameterList& p,
-                              const Teuchos::RCP<Albany::Layouts>& dl) :
-  BF        (p.get<std::string>("BF Variable Name"), dl->node_qp_scalar),
-  w_measure (p.get<std::string>("Weighted Measure Name"), dl->qp_scalar)
+                              const Teuchos::RCP<Albany::Layouts>& dl)
 {
+  std::string interpType = p.get<std::string>("Interpolation Type");
+  TEUCHOS_TEST_FOR_EXCEPTION ((interpType != "Value At Cell Barycenter") && (interpType != "Cell Average"),
```

Comment:
We could place the enum in `problems/Albany_ProblemUtils.hpp`, and force users to pass the enum itself rather than a string (less error prone). Catch typos at compile time rather than run time.

Reply:
I can do that. I just thought it would be better not to add an enum to `problems/Albany_ProblemUtils.hpp` that is only used by one evaluator.

Reply:
Yeah, that's a good point. I guess we could do either way. We might have it as a public member type in the class, then customers would have to include that file to use it. It might become a bit cumbersome, I guess.

However, since it is a binary choice, you could also think about making it a bool argument.

Anyhow, I guess there's not a ton to gain by turning the string into an enum/bool. So feel free to leave as is.

Reply:
Sorry but I'll opt for the option that requires the least effort.. i.e. leave it as is.
I don't like bools where it is not clear what the other option is...   if you have    `bool cellAverage`, it is not clear what you are doing when it is false.

Reply:
Agreed.